packages/service-utils/src/core/authorize/client.test.ts

```diff
@@ -101,4 +106,21 @@ describe("authorizeClient", () => {
     expect(result.errorCode).toBe("ORIGIN_UNAUTHORIZED");
     expect(result.status).toBe(401);
   });
+
+  it("should authorize client with incoming service api key", () => {
+    const authOptionsWithServiceKey: ClientAuthorizationPayload = {
+      secretKeyHash: null,
+      bundleId: null,
+      origin: "unauthorized.com", // Even unauthorized origin should work with service key
+      incomingServiceApiKey: "test-service-key",
+    };
+
+    const result = authorizeClient(
+      authOptionsWithServiceKey,
+      validTeamAndProjectResponse,
+      // biome-ignore lint/suspicious/noExplicitAny: test only
+    ) as any;
+    expect(result.authorized).toBe(true);
+    expect(result.project).toEqual(validTeamAndProjectResponse.project);
```

Comment:
what does it use to get the project here? was expecting a clientId in the payload above?

Reply:
ah I think the test is murky and just always returns the valid project team response here - that said the important piece here is that authorized is true despite the origin not being valid

Reply:
ah actually no, we pass in the api response to the test with `validTeamAndProjectResponse` and that's what comes back